staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -422,19 +422,50 @@ func (f *featureGate) AddVersioned(features map[Feature]VersionedSpecs) error {
 	if f.closed {
 		return fmt.Errorf("cannot add a feature gate after adding it to the flag set")
 	}
-
 	// Copy existing state
 	known := f.GetAllVersioned()
 
 	for name, specs := range features {
-		sort.Sort(specs)
 		if existingSpec, found := known[name]; found {
 			sort.Sort(existingSpec)
 			if reflect.DeepEqual(existingSpec, specs) {
 				continue
 			}
 			return fmt.Errorf("feature gate %q with different spec already exists: %v", name, existingSpec)
 		}
+		// Validate new specs are well-formed
+		var lastVersion *version.Version
+		var wasBeta, wasGA, wasDeprecated bool
+		for i, spec := range specs {
+			if spec.Version == nil {
+				return fmt.Errorf("feature %q did not provide a version", name)
+			}
```

Comment:
should this also check the spec version is only at minor granularity (`len(spec.Version.Components()) == 2`, I think), since everything using this is only handling minor boundaries for now?